src/main/scala/nl.knaw.dans.easy.bagstore/component/BagStoreComponent.scala

```diff
@@ -151,26 +152,31 @@ trait BagStoreComponent {
      * @param outputStream      the output stream to write to
      * @return whether the call was successful
      */
-    def copyToStream(itemId: ItemId, archiveStreamType: Option[ArchiveStreamType], outputStream: => OutputStream): Try[Unit] = {
+    def copyToStream(itemId: ItemId, archiveStreamType: Option[ArchiveStreamType], outputStream: => OutputStream, forceInactive: Boolean = false): Try[Unit] = {
       trace(itemId)
       val bagId = BagId(itemId.uuid)
 
       fileSystem.checkBagExists(bagId).flatMap { _ =>
         for {
           bagDir <- fileSystem.toLocation(bagId)
           itemPath <- itemId.toFileId.map(f => bagDir.resolve(f.path)).orElse(Success(bagDir))
-          _ <- validateThatFileIsActive(itemPath, itemId)
-          fileIds <- enumFiles(itemId)
+          fileIds <- enumFiles(itemId, forceInactive = true) // validation that the bagdir is active will be done later one
           fileSpecs <- createFileSpecs(bagDir, itemPath, fileIds)
           dirSpecs <- createDirectorySpecs(bagDir, itemPath, fileIds)
           allEntriesCount = fileSpecs.length + dirSpecs.length
           allEntries = () => (dirSpecs ++ fileSpecs).sortBy(_.entryPath) // only concat and sort if necessary, hence as a function here
-          _ <- archiveStreamType.map(copyToArchiveStream(itemId, allEntriesCount, outputStream)(allEntries))
+          _ <- validateBagNotHiddenAndFileIsFound(itemId, allEntriesCount, bagDir, forceInactive)
+          _ <- archiveStreamType.map(copyToArchiveStream(itemId, outputStream)(allEntries))
             .getOrElse(copyToOutputStream(itemId, fileIds, allEntriesCount, outputStream))
         } yield ()
       }
     }
 
+    private def validateBagNotHiddenAndFileIsFound(itemId: ItemId, entriesCount: Int, bagDir: BaseDir, forceInactive: Boolean): Try[Unit] = {
```

Comment:
why merge these two things into one function. The `fileIsFound` part should be a separate function to be called in the `copyToStream` for-comprehension above.

```scala
for {
    ...
    _ <- fileIsFound(allEntriesCount)
    _ <- validateThatBagDirIsNotHidden(bagDir, itemId, forceInactive)
    ...
} yield ()
```